pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -563,6 +564,13 @@ func (m *managerImpl) evictPod(pod *v1.Pod, gracePeriodOverride int64, evictMsg
 			return false
 		}
 	}
+
+	if controllerRef := metav1.GetControllerOf(pod); controllerRef != nil {
+		if controllerRef.Kind == "DaemonSet" {
```

Comment:
we should not special-case daemonset pods at this level. nothing prevents someone from making their own node-targeting controller that functions identically to daemonsets